src/Blorc.OpenIdConnect/Services/UserManager.cs

```diff
@@ -41,6 +44,40 @@ public UserManager(IJSRuntime jsRuntime, NavigationManager navigationManager, IC
 
         public Dictionary<string, object> Configuration { get; private set; }
 
+        public async Task<TUser> GetUserAsync<TUser>(Task<AuthenticationState> authenticationStateTask, JsonSerializerOptions options = null)
+        {
+            var authenticationState = await authenticationStateTask;
+            if (authenticationState.User.Identity is not null && !authenticationState.User.Identity.IsAuthenticated)
+            {
+                return default;
+            }
+
+            return await GetUserAsync<TUser>(options: options);
+        }
+
+        public async Task<TUser> GetUserAsync<TUser>(bool reload = true, JsonSerializerOptions options = null)
+        {
+            var userType = typeof(TUser);
+            if (!reload && _usersCache.TryGetValue(userType,  out var value) && value is TUser user)
+            {
+                return user;
+            }
+
+            if (reload)
```

Comment:
Why not move reload to the top? Then we don't have to check it in the if statement above.